src/common.ts

```diff
@@ -200,34 +200,133 @@ export interface ScopeError {
  * The parser interface.
  */
 export interface ParserState {
-  source: string;
-  flags: Flags;
-  index: number;
-  line: number;
-  column: number;
-  tokenPos: number;
-  startPos: number;
-  startColumn: number;
-  startLine: number;
-  colPos: number;
-  linePos: number;
-  end: number;
-  token: Token;
-  onComment: any;
-  onToken: any;
-  tokenValue: any;
-  tokenRaw: string;
+  /**
+   * The source code to be parsed
+   */
+  source: string
```

Comment:
Why remove the semicolon in interface def?

Reply:
Not an actual reason. I'm just used to don't use semis.

Reply:
But they are there for our other interface definitions. I suggest to put it back here for consistency.

I am surprised "prettier" does not have an opinion on this syntax, it accepted both. But there is `semi: true` in our `prettier.config.js`, a "prettier" bug?
If you want to cleanup the type def to remove semicolon, pls create another PR to touch all of the interfaces, if @KFlash has no strong opinion on this syntax thing.

Reply:
@fisker should not `semi:true` in `prettier.config.js` enforce the semicolon here?

Reply:
yes, why don't we run `prettier --check` on CI?

Reply:
My bad I didn't run it before the pr.

Reply:
@KFlash I am pretty sure there is git hook on pre-commit to check those thing, but I cannot find it in current master head. Did you remove the git hook?

Reply:
I tried following locally, prettier didn't pick it up.
```ts
export interface ParserState {
  source: string
  flags: Flags
  index: number;
```

```
huocp@Chunpengs-iMac ~/g/meriyah (master) [1]> npx prettier -c src/
Checking formatting...
All matched files use Prettier code style!
```

Reply:
I havent deleted anything.  The hook should be in the .git folder

Reply:
Got it, we got there in `.prettierignore`. @KFlash why we ignored those source files?
```
bench/v1.6.2/
bench/v2/
bench/v8-native-calls.js
node_modules/
dist/
src/token.ts
src/common.ts
src/chars.ts
src/lexer/charClassifier.ts
src/unicode.ts
package-lock.json
```

Reply:
the style get fucked up with Prettier and not lined. Regarding unicode.ts it will be a long list (2000 lines)

Reply:
@3cp Possible to use non-open source to replace Prettier? I got my own pretty printer we could use, but ain't going to open source that one.

Reply:
It could be weird for open source project to depend on non-open project.

Reply:
I see your point!